moldesign/integrators/lammps_integrator.py

```diff
@@ -0,0 +1,193 @@
+        self.output_iter = None     # How often we dump the positions of the atoms during the simulation
+        self.traj = None
+
+    def run(self, run_for):
+        """
+        Users won't call this directly - instead, use mol.run
+        Propagate position, momentum by a single timestep using LAMMPS NVT
+        
+        Arguments:
+            run_for (int): number of timesteps OR amount of time to run for
+            self.params.timestep (float): timestep length
+
+        """
+        self.total_iter = self.time_to_steps(run_for, self.params.timestep)
+        self.output_iter = self.time_to_steps(self.params.frame_interval, self.params.timestep)
+
+        if self.total_iter < self.output_iter:
+            raise ValueError("Run duration {0} "
+                             "can\'t be smaller than frame interval {1}".format(self.total_iter, self.output_iter))
+
+        # Prepare lammps system
+        self.prep()
+        self._configure_system()
+        lmps = self.lammps_system
+
+        predictable_filename = "dump.result"
+        dump_filepath = os.path.join(self.mol.energy_model.tmpdir, predictable_filename)
+        self.mol.energy_model.files.append(dump_filepath)
+
+        dump_command = "dump {0} all custom {1} {2} id x y z vx vy vz".format(self.NAME_RESULT,
+                                                                            self.output_iter, dump_filepath)
+
+        lmps.command(dump_command)
+        lmps.command("dump_modify {0} sort id".format(self.NAME_RESULT))
+
+        # run simulation
+        lmps.run(self.total_iter, "post no")
+
+        # Reset lammps system by undumping and unfixing
+        for dmp in lmps.dumps:
+            lmps.command("undump " + dmp.get('name'))
+
+        for fix in lmps.fixes:
+            lmps.command("unfix " + fix.get('name'))
+
+        # Create trajectory object
+        self.traj = Trajectory(self.mol, unit_system=self._model.unit_system)
+        self.mol.calculate()
+
+        # Dynamics loop over the dump file
+        dump = open(dump_filepath, "rw+")
+        content = dump.readlines()
+        for istep in xrange(self.total_iter/self.output_iter):
+            self.step(istep, content)
+            self.traj.new_frame()
+        dump.close()
+
+        # Set time
+        self.mol.time += self.time
+
+        return self.traj
+
+    def prep(self):
+        """
+        Prepare the LAMMPS system by configuring it for NVT simulation
+        
+        Arguments:
+            run_for (int): number of timesteps OR amount of time to run for
+            self.params.timestep (float): timestep length
+
+        """
+        # prepare lammps model prior to preparing the integrator
+        self._model = self.mol.energy_model
+        self._model.prep()
+
+        self.time = 0.0 * self.params.timestep
+
+        self._prepped = True
+
+    def _configure_system(self):
+        # Get lammps object from model
```

Comment:
We'll need to make sure to tell LAMMPS to handle geometry constraints. Here are the 3 things that the interface will need to handle:
 1. if `self.params.get('constrain_hbonds', False) == True`, then you'll need to set up SHAKE to constrain all bonds involving a hydrogen atom
 2. if `self.params.get('constrain_water', False) == True`, then you'll need to set up SHAKE to constrain all bond lengths in any water molecules (residue name `HOH` or `WAT`)
 3. You'll need to examine the list of constraints in `self.mol.constraints`. This is a list of objects that inherit from `GeometryConstraint` (see `moldesign/geom/constraints.py`), each of which describes a constraint that needs to be satisfied during MD.

Alternatively, you can just raise `NotImplementedError` in any/all of these cases :)